crates/env/src/api.rs

```diff
@@ -903,3 +903,56 @@ where
         instance.unlock_delegate_dependency::<E>(code_hash)
     })
 }
+
+/// Execute an XCM message locally, using the contract's address as the origin.
+///
+/// For more details consult
+/// [host function documentation](https://paritytech.github.io/substrate/master/pallet_contracts/api_doc/trait.Current.html#tymethod.xcm_execute).
+///
+/// # Errors
+///
+/// - If the message cannot be properly decoded on the pallet contracts side.
+/// - If the runtime doesn't allow for the contract unstable feature.
+/// - If the XCM execution fails because of the runtime's XCM configuration.
+///
+/// # Panics
+///
+/// Panics in the off-chain environment.
+pub fn xcm_execute<E, Call>(msg: &xcm::VersionedXcm<Call>) -> Result<()>
+where
+    E: Environment,
+    Call: scale::Encode,
+{
+    <EnvInstance as OnInstance>::on_instance(|instance| {
+        TypedEnvBackend::xcm_execute::<E, _>(instance, msg)
+    })
+}
+
+/// Send an XCM message, using the contract's address as the origin.
+///
+/// `msg` (after SCALE encoding) should be decodable to a valid instance of `RuntimeCall`
+/// enum.
```

Comment:
```suggestion
/// The `msg` argument has to be SCALE encoded, it needs to be decodable to a valid 
/// instance of the `RuntimeCall` enum.
```

And could you link "`RuntimeCall` enum" to the Rust doc of it?

Reply:
not sure there is a doc for it that's the composite enum generated for your runtime
example are here https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/reference_docs/frame_runtime_types/runtime/enum.RuntimeCall.html